src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocPackageCheck.java

```diff
@@ -63,12 +63,6 @@ public JavadocPackageCheck() {
         setFileExtensions("java");
     }
 
-    @Override
-    public void beginProcessing(String charset) {
-        super.beginProcessing(charset);
-        directoriesChecked.clear();
```

Comment:
`beginProcessing` is only called at the start of execution before we even begin processing a file. We only use 1 instance per execution, and create a new instance at the next execution, so there is nothing to clear at the start.